pkg/apis/ceph.rook.io/v1/cluster.go

```diff
@@ -53,7 +53,7 @@ func (c *CephCluster) ValidateCreate() error {
 	logger.Infof("validate create cephcluster %q", c.ObjectMeta.Name)
 	//If external mode enabled, then check if other fields are empty
 	if c.Spec.External.Enable {
-		if c.Spec.Mon != (MonSpec{}) || c.Spec.Dashboard != (DashboardSpec{}) || !reflect.DeepEqual(c.Spec.Monitoring, (MonitoringSpec{})) || c.Spec.DisruptionManagement != (DisruptionManagementSpec{}) || len(c.Spec.Mgr.Modules) > 0 || len(c.Spec.Network.Provider) > 0 || len(c.Spec.Network.Selectors) > 0 {
+		if !reflect.DeepEqual(c.Spec.Mon, MonSpec{}) || c.Spec.Dashboard != (DashboardSpec{}) || !reflect.DeepEqual(c.Spec.Monitoring, (MonitoringSpec{})) || c.Spec.DisruptionManagement != (DisruptionManagementSpec{}) || len(c.Spec.Mgr.Modules) > 0 || len(c.Spec.Network.Provider) > 0 || len(c.Spec.Network.Selectors) > 0 {
```

Comment:
Hey @ideepika I would suggest to use isZero comparison instead
```suggestion
		if reflect.ValueOf(c.Spec.Mon).isZero() || reflect.ValueOf(c.Spec.Dashboard).isZero() || reflect.ValueOf(c.Spec.Monitoring).isZero() || reflect.ValueOf(c.Spec.DisruptionManagement).isZero() || len(c.Spec.Mgr.Modules) > 0 || len(c.Spec.Network.Provider) > 0 || len(c.Spec.Network.Selectors) > 0 {
```

`isZero` checks if the struct has some value or not